packages/discovery-react-components/src/components/TooltipHighlight/TooltipHighlight.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { FC, useState, useEffect } from 'react';
+import cx from 'classnames';
+import { Tooltip } from 'carbon-components-react';
+import { settings } from 'carbon-components';
+import { TooltipAction, TooltipEvent } from './types';
+import { FacetInfoMap } from 'components/DocumentPreview/types';
+
+// TooltipInfo is the internal state of the TooltipHightlight
+interface TooltipInfo {
+  rectTooltipArea: DOMRect;
+  tooltipContent: JSX.Element;
+  isOpen: boolean;
+}
+
+type Props = {
+  /**
+   * Parent div element
+   */
+  parentDiv: React.MutableRefObject<HTMLDivElement | null>;
+
+  /**
+   * state of the highlight-tootip
+   */
+  tooltipAction: TooltipAction;
+};
+
+const Z_INDEX_ON_TOP = 50;
+const baseTooltipHighlight = `${settings.prefix}--tooltip-hightlight`;
+
+export const TooltipHighlight: FC<Props> = ({ parentDiv, tooltipAction }) => {
+  const [tooltipInfo, setTooltipInfo] = useState<TooltipInfo>({
+    rectTooltipArea: new DOMRect(),
+    tooltipContent: <div></div>,
+    isOpen: false
+  });
+
+  useEffect(() => {
+    // Compute parameters to overlay the tooltip over the active item
+    const highlightDivRect = parentDiv.current?.getBoundingClientRect() || new DOMRect();
+    const isOpen = tooltipAction.tooltipEvent !== TooltipEvent.LEAVE;
+    const clickRect = tooltipAction.rectActiveElement || new DOMRect();
+    const tooltipRect = new DOMRect(
+      clickRect.x - highlightDivRect.x,
+      clickRect.y - highlightDivRect.y,
+      clickRect?.width,
+      clickRect?.height
+    );
+    const tooltipUpdate = {
+      rectTooltipArea: tooltipRect,
+      tooltipContent: tooltipAction.tooltipContent || <div></div>,
+      isOpen: isOpen
+    };
+    setTooltipInfo(tooltipUpdate);
+  }, [tooltipAction, setTooltipInfo]);
+
+  return (
+    // Outter div is required to provide tooltip element with position information
+    // "pointerEvents" = "none" so that underlying elements can react to mouse events
+    <div
+      style={{
+        position: 'absolute',
+        zIndex: Z_INDEX_ON_TOP,
+        pointerEvents: 'none',
```

Comment:
Any reason why these shouldn't be declared in `styles`, since they're static? Then just set the class name in the outer div and add in the dynamic styles.

Reply:
moved